lib/stateTransition/AbstractStateTransition.js

```
@@ -49,6 +77,154 @@ class AbstractStateTransition {
+
+    if (extractedPublicKey !== identityPublicKey.getData()) {
+      throw new InvalidSignaturePublicKeyError(identityPublicKey.getData());
+    }
+
+    this.signaturePublicKeyId = identityPublicKey.getId();
+
+    return this;
+  }
+
+  /**
+   *
+   * @param {string} privateKey
+   * @return {AbstractStateTransition}
+   */
+  signByPrivateKey(privateKey) {
+    const data = this.serialize({ skipSignature: true });
+    const privateKeyInHex = Buffer
+      .from(privateKey, 'base64')
+      .toString('hex');
+
+    this.signature = signData(data, privateKeyInHex).toString('base64');
+
+    return this;
+  }
+
+  /**
+   * Verify signature
+   *
+   * @param {IdentityPublicKey} publicKey
+   * @return {boolean}
+   */
+  verifySignature(publicKey) {
+    const signature = this.getSignature();
+    if (!signature) {
+      return false;
+    }
+
+    const data = this.serialize({ skipSignature: true });
+
+    try {
+      const publicKeyBuffer = Buffer.from(publicKey.getData(), 'base64');
+      // eslint-disable-next-line no-underscore-dangle
+      const publicKeyHash = new PublicKey(publicKeyBuffer)._getID();
+
+      return verifyDataSignature(
+        data,
+        Buffer.from(signature, 'base64'),
+        publicKeyHash,
+      );
+    } catch (e) {
+      return false;
+    }
+  }
+
+  /**
+   * Verify signature with public key
+   * @param {string} publicKey
+   * @returns {boolean}
+   */
+  verifySignatureByPublicKey(publicKey) {
+    const signature = this.getSignature();
+    if (!signature) {
+      return false;
+    }
+    const data = this.serialize({ skipSignature: true });
+
+    try {
+      const publicKeyBuffer = Buffer.from(publicKey, 'base64');
+
+      // eslint-disable-next-line no-underscore-dangle
+      const publicKeyHash = new PublicKey(publicKeyBuffer)._getID();
+
+      return verifyDataSignature(
+        data,
+        Buffer.from(signature, 'base64'),
+        publicKeyHash,
+      );
+    } catch (e) {
```

Not sure that it's a good idea. We just hiding errors and make debugging much more difficult.